esmvalcore/preprocessor/_area.py

```diff
@@ -295,3 +302,86 @@ def extract_named_regions(cube, regions):
     constraints = iris.Constraint(region=lambda r: r in regions)
     cube = cube.extract(constraint=constraints)
     return cube
+
+
+def _clip_geometries(cube, geometries):
+    """Clip cube to only include the geometries and padding."""
```

Comment:
a bit confused as to why to include `geometries` as a functional argument when it's actually used only in the main function `extract_shape`

Reply:
It is used in line 317 to get the bounds for the clipping box. I suppose we could also just pass geometries.bounds, since it's the only thing we need.